src/bottles/Operation.java

```diff
@@ -6,67 +6,66 @@
 public class Operation {
     public static void main(String[] args) {
 
-        int numOfBottles = 1;
+        int numOfBottles = 0, amountOfLiters, bottleId, bottleIdOut, bottleIdIn;
         Bottle[] bottles;
-        boolean isNumOfBottlesRight;
+        boolean isValid;
+        String choiceAction;
+        String menu = ("\nChoose operation:\n1. Pour in\n2. Pour out\n3. Transfer\n4. Show bottles data\n5. Exit");
 
         do {
             try {
                 System.out.print("Enter number of bottles: ");
                 Scanner sc = new Scanner(System.in);
                 numOfBottles = sc.nextInt();
-                isNumOfBottlesRight = true;
-
-            }
-            catch (InputMismatchException e) {
-                System.out.println("Invalid quantity specified");
-                isNumOfBottlesRight = false;
             }
+            catch (InputMismatchException e) {}
 
-            if (numOfBottles <= 0) {
+            isValid = numOfBottles > 0;
+            if (!isValid) {
                 System.out.println("Invalid quantity specified");
-                isNumOfBottlesRight = false;
-                numOfBottles = 1;
             }
 
         }
-        while (!isNumOfBottlesRight);
-
+        while (!isValid);
 
         // random capacity and fill level of bottles
         bottles = Bottle.createFillBottle(numOfBottles);
 
-        //Bottle.bottleSettings();
-
-
-
-        //void bottleSettings() {
-//
-        //}
-
-
-
-
-
-        System.out.println("\nChoose operation: \n1. Sour in\n2. Sour out \n3. Transfer\n4. Show bottles data \n5. Exit");
-
-        try {
+        do {
+            System.out.println(menu);
             Scanner sc = new Scanner(System.in);
-            int choiceOperation = sc.nextInt();
-        }
-        catch (InputMismatchException e) {
-            System.out.println("error");
+            choiceAction = sc.nextLine();
+
+            switch (choiceAction) {
+                case "1":
+                    bottleId = Bottle.whichBottle(bottles);
+                    amountOfLiters = Bottle.howManyLiters("pour in");
+                    Bottle.pourIn(bottles[bottleId - 1], amountOfLiters);
+                    break;
+                case "2":
+                    bottleId = Bottle.whichBottle(bottles);
+                    amountOfLiters = Bottle.howManyLiters("pour out");
+                    Bottle.pourOut(bottles[bottleId - 1], amountOfLiters);
+                    break;
+                case "3":
+                    System.out.print("From which bottle pour the liquid. ");
+                    bottleIdOut = Bottle.whichBottle(bottles);
+                    System.out.print("To which bottle pour the liquid. ");
+                    bottleIdIn = Bottle.whichBottle(bottles);
+                    amountOfLiters = Bottle.howManyLiters("transfer");
+                    Bottle.transfer(bottles[bottleIdOut - 1], bottles[bottleIdIn -1], amountOfLiters);
```

Comment:
Here you have good example why mixing logic of `ID` with `Index` can cause a problem.

Lets assume that you don't know what is the implementation of a function, but you have documentation of it.
I would say it could be something like this:
```java
/// Ask user to select bottle from given bottles.
/// - Parameters: 
///   - bottle: List of available bottles to select from.
/// - Returns: Selected bottle Id 
static int Bottle.whichBottle(Bottle[] bottle)
```
Then you can not use like it this: `bottles[bottleIdOut - 1]`, because `bottleIdOut - 1` is not an Id of a bottle but an Index. 

Well, in this case it might be true that Index is just `id - 1`, because Id is created from 1 to the number of bottles and always resets when program is relaunched. However if you will try to extend functionality, for example to store bottles or whatever, then you will have to change every single function, because everywhere "code" knows how other parts of application are implemented.

How to solve this? Easy. If you will decide to return `Bottle.id`, then select bottle by `id`. If you will return `Index` of bottle, then select bottle by an Index. Or return `Bottle` and you will not have to select it again 👍